src/main/scala/mesosphere/marathon/core/storage/store/impl/cache/LazyCachingPersistenceStore.scala

```diff
@@ -28,13 +29,16 @@ import scala.concurrent.{ ExecutionContext, Future }
   * @tparam Serialized The serialized format for the persistence store.
   */
 class LazyCachingPersistenceStore[K, Category, Serialized](
-    val store: BasePersistenceStore[K, Category, Serialized])(implicit
+    val store: BasePersistenceStore[K, Category, Serialized],
+    val versionedValueCacheConfig: LazyCachingPersistenceStore.VersionedValueCacheConfig)(implicit
   mat: Materializer,
     ctx: ExecutionContext) extends PersistenceStore[K, Category, Serialized] with StrictLogging {
 
   private val lockManager = LockManager.create()
   private[store] val idCache = TrieMap.empty[Category, Seq[Any]]
   private[store] val valueCache = TrieMap.empty[K, Option[Any]]
+  private[store] val versionCache = TrieMap.empty[(Category, K), Seq[Any]]
```

Comment:
`idCache` is "Category -> MarathonId", `valueCache` is "MarathonId -> Option[Some Value]".

`versionCache` is "Category, MarathonId" -> what, a list of versions? If that's the case, it should be `Seq[OffsetDateTime]` as the valuetype.